preview_generator/preview/builder/image__imconvert.py

```diff
@@ -135,3 +148,40 @@ def build_jpeg_preview(
         return ImagePreviewBuilderPillow().build_jpeg_preview(
             tmp_filepath, preview_name, cache_path, page_id, extension, size
         )
+
+    def _imagemagick_convert(self, source_path: str, dest_path: str, mimetype: str = "") -> int:
```

Comment:
if `mimetype` default value purpose is to say "no mimetype given", prefer use `None` and test `is None` in code. This way prevent from hiding case where `mimetype` is str empty by error.

Reply:
i suggest to add ` assert mimetype != ''" ` empty mimetype string should not happened at all and is not supported properly

Reply:
is the current fix  ok ?

Reply:
Ok for `mimetype: typing.Optional[str] = None`